src/include/duckdb/common/types/vector_buffer.hpp

```diff
@@ -48,6 +50,7 @@ class VectorBuffer {
 	}
 	VectorBuffer() {
 	}
+	shared_ptr<ArrowArrayWrapper> arrow_array;
```

Comment:
Could we make this more generic? I'm thinking about having a generic struct, e.g.:


```cpp
struct VectorAuxiliaryData {
   virtual ~ VectorAuxiliaryData();
};
```

The VectorBuffer can then have a `unique_ptr< VectorAuxiliaryData> extra_data;` as a member. Inside the Arrow integration, you can then inherit from the `VectorAuxiliaryData`, and create a `ArrowVectorAuxiliaryData` object.

This way we (1) don't need to include the arrow wrapper stuff everywhere the vector_buffer is included, and (2) can leave this as a more generic solution that others can also use.

Reply:
Yes that was also my concern here